Elevel/Elevel.Infrastructure.Services/Implementation/FileService.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Elevel.Application.Interfaces;
+using Elevel.Domain.Models;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+
+namespace Elevel.Infrastructure.Services.Implementation
+{
+    public class FileService : IFileService
+    {
+        private readonly IWebHostEnvironment _hostingEnvironment;
+
+        public FileService(IWebHostEnvironment hostingEnvironment)
+        {
+            _hostingEnvironment = hostingEnvironment;
+        }
+
+        public void UploadFiles(List<IFormFile> files)
+        {
+            var target = @"wwwroot\files";
+            Directory.CreateDirectory(target);
+            files.ForEach(async file =>
+            {
+                if (file.Length <= 0) return;
+                var filePath = "example.txt";
+                for (int i = 0; ; ++i)
+                {
+                    var fileName = Path.GetFileNameWithoutExtension(file.FileName);
+                    var fileExtension = Path.GetExtension(file.FileName);
+                    filePath = Path.Combine(target, fileName + (i > 0 ? $" ({i})" : "") + fileExtension);
+                    if (!File.Exists(filePath)) break;
+                }
+                using (var stream = new FileStream(filePath, FileMode.Create))
+                {
+                    await file.CopyToAsync(stream);
+                }
+            });
+        }
+
+        public FileZip DownloadFiles(string fromDirectory)
```

Comment:
instead fromDirectory needs fromFilePath
For example "myFile2021_02_19_16_00_50.mp3"